S390x update builder image

AlekseiNikiforovIBM opened the pull request:
Publish current state of s390x builder image to allow reproducing worker setup.
Also, if this image gets published to docker repository later, it'd be possible to download published image instead of building it into worker image in https://github.com/pytorch/pytorch/blob/main/.github/scripts/s390x-ci/self-hosted-builder/actions-runner.Dockerfile#L66, which should allow improving restart time at the cost of additional runtime overhead.

Compared to first attempt to merge:
- default docker repository settings are added to all runners. Changes are mirrored in this PR.
- job is moved into separate workflow file.
- it's no longer attempted to update limits on s390x. Limits should be properly set up there on the host. And it's not possible to update them from worker since it runs in container. Also, worker container currently doesn't have sudo installed or configured or any systemd running.
- github token is now passed once via named pipe instead of environment variable. This should increase security of tokens.

AlekseiNikiforovIBM committed:
s390x: migrate to manylinux

AlekseiNikiforovIBM committed:
Rebuild all python versions

AlekseiNikiforovIBM committed:
Reset entrypoint and initial cmd

AlekseiNikiforovIBM committed:
Install cmake

AlekseiNikiforovIBM committed:
Switch to python 3.12 and update all dependencies

AlekseiNikiforovIBM committed:
Add test build dependencies

AlekseiNikiforovIBM committed:
Install periodic tests' dependencies

AlekseiNikiforovIBM committed:
Install more test dependencies

AlekseiNikiforovIBM committed:
Update GPU_IMAGE for s390x image

The value is still unused

AlekseiNikiforovIBM committed:
Install sudo into s390x image

It is used in openssl build script

AlekseiNikiforovIBM committed:
Fix paths

AlekseiNikiforovIBM committed:
Also fix building images for non-s390x

It looks like there are no such packages anymore in repositories

AlekseiNikiforovIBM committed:
Don't rebuild openssl

Current manylinux has same version,
but due to some different build options,
custom-built openssl breaks system after some updates
due to missing symbols.

AlekseiNikiforovIBM commented:
@pytorchbot rebase

AlekseiNikiforovIBM commented:
@malfet , could you please review this change?

I didn't test WITH_PUSH=true.
Also, I'm not using "pytorch/test-infra/.github/actions/calculate-docker-image" due to it relying on AWS credentials. Instead I use ".ci/docker/manywheel/build.sh manylinuxs390x-builder:cpu-s390x" for both WITH_PUSH=true and WITH_PUSH=false.

huydhn reviewed (approved):
LGTM!

malfet commented:
@pytorchbot merge -f "S390 docker build (https://github.com/pytorch/pytorch/actions/runs/10680686839/job/29602702979?pr=132983 ) and lint are green "

atalman commented:
@pytorchmergebot revert -c nosignal -m "Authenticate With PUSH is failing. Error: no registries found in registries.conf, a registry must be provided. Error: Process completed with exit code 125."

AlekseiNikiforovIBM commented:
> @pytorchmergebot revert -c nosignal -m "Authenticate With PUSH is failing. Error: no registries found in registries.conf, a registry must be provided. Error: Process completed with exit code 125."

I guess WITH_PUSH failed. As I wrote earlier, I didn't test this scenario. Is there a log publicly available? Otherwise, I can try disabling WITH_PUSH=true completely.

AlekseiNikiforovIBM committed:
Build test some test dependencies in advance

Build test dependencies in advance
if they require some special handling (patches, special settings, etc)

malfet commented:
> > @pytorchmergebot revert -c nosignal -m "Authenticate With PUSH is failing. Error: no registries found in registries.conf, a registry must be provided. Error: Process completed with exit code 125."
> 
> I guess WITH_PUSH failed. As I wrote earlier, I didn't test this scenario. Is there a log publicly available? Otherwise, I can try disabling WITH_PUSH=true completely.

Yes, all failures are public, for example one can see it on the commit that were merged, see https://github.com/pytorch/pytorch/actions/runs/10963649935/job/30445556004

AlekseiNikiforovIBM committed:
Add podman settings to behave closer to docker

AlekseiNikiforovIBM committed:
Don't update limits on s390x runners

AlekseiNikiforovIBM committed:
Move docker image build job into separate workflow

malfet commented:
> Could you please take a look at this PR again? I'd like to have it merged again. If new unexpected issues would reappear, it's ok for me to revert it once more.

@AlekseiNikiforovIBM  can you please update PR description so that one can understand better what have changed since the last merge attempt?

malfet commented:
> > Could you please take a look at this PR again? I'd like to have it merged again. If new unexpected issues would reappear, it's ok for me to revert it once more.
> 
> @AlekseiNikiforovIBM can you please update PR description so that one can understand better what have changed since the last merge attempt? And in general PR description is a bit too vague, unsure what you are trying to achieve by it

AlekseiNikiforovIBM commented:
> > > Could you please take a look at this PR again? I'd like to have it merged again. If new unexpected issues would reappear, it's ok for me to revert it once more.
> > 
> > 
> > @AlekseiNikiforovIBM can you please update PR description so that one can understand better what have changed since the last merge attempt? And in general PR description is a bit too vague, unsure what you are trying to achieve by it

Thanks, I've updated description. Please let me know if it needs more improvements.

AlekseiNikiforovIBM committed:
Pass github runner token through pipe

malfet commented:
@pytorchbot merge
